src/fetch.ts

```diff
@@ -21,6 +22,16 @@ const RETRIABLE_ERRORS: string[] = [
 export interface PayloadWithOrigin extends JsonRpcRequest {
   origin?: string;
 }
+
+/**
+ * Like a JSON-RPC request, but includes an optional `origin` property.
+ * This will be included in the request as a header if specified.
+ */
+export type JsonRpcRequestWithOrigin<Params extends JsonRpcParams> =
```

Comment:
Purposefully not reusing `PayloadWithOrigin` above because 1) it's badly named, 2) it doesn't take any type parameters and 3) we need a type and not an interface going forward.

Reply:
Do you think we should mark `PayloadWithOrigin` as deprecated or do you see it still having some use case in the future?

Reply:
Good point! I have done so in [12ec424](https://github.com/MetaMask/eth-json-rpc-middleware/pull/357/commits/12ec42419ce293c7370e140756f02d5678d57cab).